src/main/scala/org/clulab/habitus/utils/JsonPrinter.scala

```diff
@@ -40,47 +39,21 @@ class JsonPrinter(outputFilename: String) extends Printer {
           valueMention.lemmas
               .map(_.mkString(" "))
               .getOrElse(valueMention.text)
-    val jObject = contextDetailsMap
-        .get(mention.sentence) // Get it, optionally, if it is there.
-        .map { contextDetails => // If there, do this.
-          ("variableText" -> variableText) ~
-          ("valueText" -> valueText) ~
-          ("valueNorm" -> valueNorm) ~
-          ("sentenceText" -> sentenceText) ~
-          ("inputFilename" -> inputFilename) ~
-          ("mostFreqLoc0Sent" -> contextDetails.mostFreqLoc0Sent) ~
-          ("mostFreqLoc1Sent" -> contextDetails.mostFreqLoc1Sent) ~
-          ("mostFreqLoc" -> contextDetails.mostFreqLoc) ~
-          ("mostFreqDate0Sent" -> contextDetails.mostFreqDate0Sent) ~
-          ("mostFreqDate1Sent" -> contextDetails.mostFreqDate1Sent) ~
-          ("mostFreqDate" -> contextDetails.mostFreqDate) ~
-          ("mostFreqCrop0Sent" -> contextDetails.mostFreqCrop0Sent) ~
-          ("mostFreqCrop1Sent" -> contextDetails.mostFreqCrop1Sent) ~
-          ("mostFreqCrop" -> contextDetails.mostFreqCrop) ~
-          ("mostFreqFert0Sent" -> contextDetails.mostFreqFertilizer0Sent) ~
-          ("mostFreqFert1Sent" -> contextDetails.mostFreqFertilizer1Sent) ~
-          ("mostFreqFert" -> contextDetails.mostFreqFertilizerOverall)
-        }
-        .getOrElse { // If it wasn't there, do this instead.
-          // These keys should match the ones used above.
-          ("variableText" -> variableText) ~
-            ("valueText" -> valueText) ~
-            ("valueNorm" -> valueNorm) ~           
-            ("sentenceText" -> sentenceText) ~
-            ("inputFilename" -> inputFilename) ~
-            ("mostFreqLoc0Sent" -> "N/A") ~
-            ("mostFreqLoc1Sent" -> "N/A") ~
-            ("mostFreqLoc" -> "N/A") ~
-            ("mostFreqDate0Sent" -> "N/A") ~
-            ("mostFreqDate1Sent" -> "N/A") ~
-            ("mostFreqDate" -> "N/A") ~
-            ("mostFreqCrop0Sent" -> "N/A") ~
-            ("mostFreqCrop1Sent" -> "N/A") ~
-            ("mostFreqCrop" -> "N/A") ~
-            ("mostFreqFert0Sent" -> "N/A") ~
-            ("mostFreqFert1Sent" -> "N/A") ~
-            ("mostFreqFert" -> "N/A")
-        }
+
+    var jObject: JObject =
+      ("variableText" -> variableText) ~
+        ("valueText" -> valueText) ~
+        ("valueNorm" -> valueNorm) ~
+        ("sentenceText" -> sentenceText) ~
+        ("inputFilename" -> inputFilename)
+
+    // add key-value pairs from the context attachment
+    mention.attachments.head.asInstanceOf[Context].getArgValuePairs()
+      .foreach {
+        i =>
```

Comment:
I've been thinking about this.  Isn't a side-effect of `getArgValuePairs` and then `toString` that the results are all strings, even if they are numbers like context.comparative might be?  If so, it might fool a consumer of the output.

I'm not happy with my code below, but something like it might be necessary:
```scala
    // This should be reusable across JPrinters.
    def toJObject(argValuePairs: List[(String, AnyRef)]): JObject = {
      new JObject(argValuePairs.map { case (arg, value) => JField(arg,
        value match {
          case l: java.lang.Long => JLong(l)
          case i: java.lang.Integer => JInt(BigInt(i))
          case s: String => JString(s)
        }
      )})
    }

    val jObject1: JObject =
      ("variableText" -> variableText) ~
        ("valueText" -> valueText) ~
        ("valueNorm" -> valueNorm) ~
        ("sentenceText" -> sentenceText) ~
        ("inputFilename" -> inputFilename)
    val jObject2: JObject = toJObject(mention.attachments.head.asInstanceOf[Context].getArgValuePairs())
    val jObject = jObject1 ++ jObject2
```

Reply:
> If so, it might fool a consumer of the output.

Great point!

Reply:
Hm, looks like adding objects like that results in a different json structure, e.g.,

```
[ {
    "variableText" : "area",
    "valueText" : "35%",
    "valueNorm" : "35 %",
    "sentenceText" : "Financing of agricultural production  Credit committees ( 1er and 2th ) of La Banque Agricole Zone Nord met and , the volume of credit retained for the Financing of  Agricultural Production for the 2021_2022 rainy season , amounts to 3.635 billion CFA francs corresponding to an area equal to 11 268  hectares , i.e. a decrease of 35 % compared to the overall amount of credit granted to finance the production of the  wintering campaign 2020/2021 .",
    "inputFilename" : "Bulletin_24_08_2021.fr.en.txt"
  }, {
    "location" : "2021_2022",
    "date" : "north zone",
    "process" : "credit",
    "crop" : "yam",
    "fertilizer" : "N/A",
    "comparative" : 0
  } ]
```

Reply:
Never mind, after using toJObject, I do this and it works fine:
```
for (value <- argJObject.obj) {
      jObject = jObject ~ (value._1 -> value._2)
    }
```